### DISCUSSION TITLE
Multiple clusters - multiple image policies?

### DISCUSSION THREAD

**Discussion by emilbillberg** [fluxcd/flux2#2666] [Q&A] [answered]
Hi,

I tried to look for an answer in the docs and in Github discussions but couldn't find anything about what the recommendations are for image policies in a multiple cluster setup. We have multiple production clusters managed by Flux. We are using image policies and image update automation controller to trigger new deploys. Right now we have one image policy per app in each production cluster and they look identical. They are all triggered on the same tag. So x clusters will create x commits in the repo when deploying. Would it make sense to only have one image policy in one cluster to handle the image automation controller and only have one commit in Github. I guess one drawback to that solution would be if you need to rollback in one of the production clusters - you would be forced to rollback in all of them. I guess also if the cluster with image policies would have downtime the other clusters would be affected. Any other cons or pros? What are the recommendations?

Thanks for a great project!

**Comment by stefanprodan** [comment, marked as answer]
Flux will push a single commit no matter on how many clusters it runs, the fastest cluster will push the changes, then all others will see there is nothing to commit and do nothing.

**Reply by emilbillberg** [reply]
Ok that makes sense that you would only get one commit. I guess we used different names for the image policies in each production cluster so they would actually create different commits. Would make more sense to have the same name to avoid that. In that case, I don't see any downsides to have equal image policies in multiple clusters.
Thanks for feedback!
